src/main/java/service/impl/FileReaderImpl.java

```diff
@@ -0,0 +1,25 @@
+package service.impl;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import service.Reader;
+
+public class FileReaderImpl implements Reader {
+    private final List<String> listString = new ArrayList<>();
```

Comment:
it shouldn't be a field, it can only cause bugs if you try to use reader two times in a row, should be a local variable, moreover, listString is a bad name. Don't name List<> list, []array, map map etc, it doesn't add any new valuable information, that could make the code more readable, remember that variable naming is an opportunity to make code understandable for someone who will read it for the first time, devs spend way more time reading code written by others, than writing their own